### PR TITLE
fix(ci): fix breaking changes brought by upload/download artifacts version update

### PR DESCRIPTION
**Pull request by paul-oureib**
## Description

**Fixes** # MON-34390

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/perl-cpan-libraries.yml [hunk 1]
```diff
@@ -345,14 +354,61 @@ jobs:
...
         with:
-          name: packages-${{ matrix.package_extension }}-${{ matrix.distrib }}
+          name: packages-${{ matrix.package_extension }}-${{ matrix.distrib }}-${{ steps.package-name.outputs.name_with_dash}}
           path: ./*.${{ matrix.package_extension }}
           retention-days: 1
 
-  sign-rpm:
+  merge-package-rpm-artifacts:
     needs: [package-rpm]
+    if: always()
```
Comment (kduret) [inline .github/workflows/perl-cpan-libraries.yml:374]:
```suggestion
```

File: .github/workflows/perl-cpan-libraries.yml [hunk 2]
```diff
@@ -345,14 +354,61 @@ jobs:
...
+        uses: ./.github/actions/merge-artifacts
+        with:
+          target_name: packages-rpm-${{ matrix.distrib }}
+          source_paths: packages-rpm-${{ matrix.distrib }}/*.rpm
+          source_name_pattern: packages-rpm-${{ matrix.distrib }}-
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  merge-package-deb-artifacts:
+    needs: [package-deb]
+    if: always()
```
Comment (kduret) [inline .github/workflows/perl-cpan-libraries.yml:393]:
```suggestion
```

File: .github/workflows/perl-cpan-libraries.yml [hunk 3]
```diff
@@ -389,13 +445,13 @@ jobs:
           key: ${{ github.sha }}-${{ github.run_id }}-rpm-${{ matrix.distrib }}
 
   download-and-cache-deb:
-    needs: [package-deb]
+    needs: [package-deb, merge-package-deb-artifacts]
```
Comment (kduret) [inline .github/workflows/perl-cpan-libraries.yml:448]:
```suggestion
    needs: [merge-package-deb-artifacts]
```

File: .github/workflows/perl-cpan-libraries.yml [hunk 4]
```diff
@@ -345,14 +354,61 @@ jobs:
...
+      - name: Merging Artifacts
+        uses: ./.github/actions/merge-artifacts
+        with:
+          target_name: packages-deb-${{ matrix.distrib }}
+          source_paths: packages-deb-${{ matrix.distrib }}/*.deb
+          source_name_pattern: packages-deb-${{ matrix.distrib }}-
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  sign-rpm:
+    needs: [package-rpm, merge-package-rpm-artifacts]
```
Comment (kduret) [inline .github/workflows/perl-cpan-libraries.yml:411]:
```suggestion
    needs: [merge-package-rpm-artifacts]
```

**Comment by paul-oureib** [comment]
tysm @a-eljazouly :)

**Pull request merged**